client.go

```diff
@@ -93,9 +107,44 @@ func (c *Client) Query(ctx context.Context, cmd string, args ...interface{}) Arg
 	}
 
 	r, err := c.Do(&Request{
+		Addr: addr,
+		Cmds: []Command{{
+			Cmd:  cmd,
+			Args: List(args...),
+		}},
+		Context: ctx,
+	})
+	if err != nil {
+		return newArgsError(err)
+	}
+
+	return r.Args
+}
+
+// MultiQuery issues a request with multiple Command to the Redis server at
+// the address set on the client and returns the response's Args.
+//
+// The commands "MULTI" and "EXEC" will be automatically added to the list of
+// Commands.
+//
+// cmds is a list of Command (see request.go)
+//
+// The context passed as first argument allows the operation to be canceled
+// asynchronously.
+func (c *Client) MultiQuery(ctx context.Context, cmds ...Command) Args {
+	addr := c.Addr
+	if len(addr) == 0 {
+		addr = "localhost:6379"
+	}
+
```

Comment:
Let's check that the list of command doesn't contain any `MULTI`, `EXEC` or `DISCARD`.